tests/test_detect_db_type_and_version.py

```
@@ -0,0 +1,231 @@
+            result = client._server.execute("SELECT 1 as test")
+            assert result is not None
+            
+            # Test detect_db_type_and_version
+            db_type, version = client._server.detect_db_type_and_version()
+            
+            # Verify results
+            assert db_type == "seekdb"
+            assert version is not None
+            assert isinstance(version, Version)
+            # Test version comparison
+            assert version > Version("0.0.0.0"), f"Version should be greater than 0.0.0.0, got: {version}"
```

Ruff (0.15.2)

[error] Possible binding to all interfaces

(S104)